File: DAY01/src/main/java/com/example/tutorial2/MainActivity.java
package com.example.tutorial2;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void addClick(View v) {
        EditText number1 = (EditText) findViewById(R.id.number1); // number1 값
        EditText number2 = (EditText) findViewById(R.id.number2); // number2 값
        TextView result = (TextView) findViewById(R.id.result); // 결과 값
        int n1 = Integer.parseInt(number1.getText().toString()); // 정수형으로 저장
        int n2 = Integer.parseInt(number2.getText().toString()); // 정수형으로 저장
        result.setText(Integer.toString(n1 + n2)); // 덧셈 결과 출력
    }

    public void subtractClick(View v) {
        EditText number1 = (EditText) findViewById(R.id.number1); // number1 값
        EditText number2 = (EditText) findViewById(R.id.number2); // number2 값
        TextView result = (TextView) findViewById(R.id.result); // 결과 값
        int n1 = Integer.parseInt(number1.getText().toString()); // 정수형으로 저장
        int n2 = Integer.parseInt(number2.getText().toString()); // 정수형으로 저장
        result.setText(Integer.toString(n1 - n2)); // 뺄셈 결과 출력
    }

    public void multiplyClick(View v) {
        EditText number1 = (EditText) findViewById(R.id.number1); // number1 값
        EditText number2 = (EditText) findViewById(R.id.number2); // number2 값
        TextView result = (TextView) findViewById(R.id.result); // 결과 값
        int n1 = Integer.parseInt(number1.getText().toString()); // 정수형으로 저장
        int n2 = Integer.parseInt(number2.getText().toString()); // 정수형으로 저장
        result.setText(Integer.toString(n1 * n2)); // 곱셈 결과 출력
    }

    public void divideClick(View v) {
        EditText number1 = (EditText) findViewById(R.id.number1); // number1 값
        EditText number2 = (EditText) findViewById(R.id.number2); // number2 값
        TextView result = (TextView) findViewById(R.id.result); // 결과 값
        int n1 = Integer.parseInt(number1.getText().toString()); // 정수형으로 저장
        int n2 = Integer.parseInt(number2.getText().toString()); // 정수형으로 저장
        result.setText(Integer.toString(n1 / n2)); // 나눗셈 결과 출력
    }

} 

File: QRScannerProject/src/main/java/com/example/qrscanner/MainActivity.kt
package com.example.qrscanner

import android.content.Intent
import android.graphics.BitmapFactory
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import com.google.zxing.integration.android.IntentIntegrator
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    // QR코드리더 버튼을 클릭했을 때 실행
    fun startBarcodeReader(view: View) {
        // 바코드 스캐너가 실행
        IntentIntegrator(this).initiateScan()
    }

    fun startBarcodeReaderCustom(view: View) {
        val integrator = IntentIntegrator(this)
        integrator.setDesiredBarcodeFormats(IntentIntegrator.QR_CODE) // 원하는 규격 설정 가능
        integrator.setPrompt("QR 코드를 스캔하여 주세요") // 스캔 시 하단 문구 설정
        integrator.setCameraId(0) // 후면 카메라 (1은 전면 카메라)
        integrator.setBeepEnabled(true) // 바코드 인식 시 Beep 소리 발생
        integrator.setBarcodeImageEnabled(true) // OnActivityResult에 바코드에 대한 결과값과 사진도 함께 전달
        integrator.initiateScan()
    }

    fun startBarcodeReaderCustomActivity(view: View) {
        val integrator = IntentIntegrator(this)
        integrator.setBarcodeImageEnabled(true) // OnActivityResult에 바코드에 대한 결과값과 사진도 함께 전달
        integrator.captureActivity = MyBarcodeReaderActivity::class.java // 클래스 전달
        integrator.initiateScan()
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        val result = IntentIntegrator.parseActivityResult(requestCode, resultCode, data)
        if (result != null) {
            if (result.contents != null) {
                Toast.makeText(this, "scannerd: ${result.contents} format: ${result.formatName}", Toast.LENGTH_LONG).show()
            } else {
                Toast.makeText(this, "Cancelled", Toast.LENGTH_LONG).show()
            }
            if (result.barcodeImagePath != null) {
                val bitmap = BitmapFactory.decodeFile(result.barcodeImagePath) // 바코드 이미지 경로 전달
                scannedBitmap.setImageBitmap(bitmap)
            }
        } else {
            super.onActivityResult(requestCode, resultCode, data)
        }
    }
}